routes/articles.js

```diff
@@ -0,0 +1,129 @@
+const express = require('express')
+let router = express.Router()
+const Article = require('../models/article')
+const {isLoggedIn, checkArticleOwnership} = require('../middleware')
+
+function escapeRegex(text) {
+    return text.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, "\\$&");
+}
+
+//INDEX ROUTE -- Show all articles
+router.get('/', (req, res) => {
+    if(req.query.search && req.xhr) {
+        const regex = new RegExp(escapeRegex(req.query.search), 'gi')
```

Comment:
Is there a possibility, that the user could input harmful regex, i.e. one that would DDoS the server?

Reply:
I'll create another issue for security concerns that will be addressed in another PR

Reply:
Issue #13